packages/vite-plugin-cloudflare/src/index.ts

```diff
@@ -416,35 +416,34 @@ export function cloudflare(pluginConfig: PluginConfig = {}): vite.Plugin[] {
 			},
 			async resolveId(source, importer, options) {
 				// Handle the virtual modules that come from Node.js compat aliases.
-				const from = maybeStripNodeJsVirtualPrefix(source);
-				if (!from) {
+				const strippedSource = maybeStripNodeJsVirtualPrefix(source);
+				if (!strippedSource) {
 					return;
 				}
 
 				const workerConfig = getWorkerConfig(this.environment.name);
 				if (!isNodeCompat(workerConfig)) {
-					return this.resolve(from, importer, options);
+					// We are not in Node.js compat mode, so we must not try to apply any unenv aliases.
+					// Just resolve the module id as normal.
+					return this.resolve(strippedSource, importer, options);
 				}
 
-				const unresolvedAlias = dealiasVirtualNodeJSImport(from);
-				const resolvedAlias = await this.resolve(
-					unresolvedAlias,
-					import.meta.url
-				);
-				assert(
-					resolvedAlias,
-					"Failed to resolve aliased nodejs import: " + unresolvedAlias
-				);
+				// Resolve this id following any unenv aliases.
+				const { unresolved, resolved } = resolveNodeJSImport(strippedSource);
 
 				if (this.environment.mode === "dev" && this.environment.depsOptimizer) {
 					// Make sure the dependency optimizer is aware of this aliased import
-					this.environment.depsOptimizer.registerMissingImport(
-						unresolvedAlias,
-						resolvedAlias.id
-					);
+					const optimized =
+						this.environment.depsOptimizer.registerMissingImport(
+							unresolved,
+							resolved
+						).id;
+					// We must pass the id from the depsOptimizer through the rest of the
+					// resolving pipeline to ensure that the optimized version gets used.
+					return this.resolve(optimized);
 				}
 
-				return resolvedAlias;
+				return resolved;
```

Comment:
Not sure it we should pass this through the `this.resolve()` call too, but since it is going to be an absolute path at this stage, I assume that no other plugin would do anything with it anyway?

Reply:
I passed it through `this.resolve()` and the tests pass.